kratos/solving_strategies/convergencecriterias/convergence_criteria.h

```diff
@@ -437,6 +457,29 @@ class ConvergenceCriteria
     ///@name Protected Operations
     ///@{
 
+    /**
+     * @brief This method validate and assign default parameters
+     * @param rParameters Parameters to be validated
+     * @param DefaultParameters The default parameters
+     */
+    virtual void ValidateAndAssignParameters(
+        Parameters& rParameters,
+        const Parameters DefaultParameters
+        ) const
```

Comment:
```suggestion
    void ValidateAndAssignParameters(
        Parameters rParameters,
        const Parameters DefaultParameters
        ) const
```

Reply:
I would keep the virtual in case we need to do some special validation. Can anyone remember why Parameter cannot be referred?